community/detectors/papercut_ng_mf_CVE_2023_27350/src/main/java/com/google/tsunami/plugins/papercut/PapercutNGMFVulnDetector.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.papercut;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.*;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.*;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+
+@PluginInfo(
+        type = PluginType.VULN_DETECTION,
+        name = "PapercutNGMRVulnDetectorWithPayload",
+        version = "1.0",
+        description = "Detects papercut versions that are vulnerable to authentication bypass and RCE.",
+        author = "Isaac_GC (isaac@nu-that.us)",
+        // How should Tsunami scanner bootstrap your plugin.
+        bootstrapModule = PapercutNGMFVulnDetectorBootstrapModule.class)
+// Optionally, each VulnDetector can be annotated by service filtering annotations. For example, if
+// the VulnDetector should only be executed when the scan target is running Jenkins, then add the
+// following @ForSoftware annotation.
+// @ForSoftware(name = "Jenkins")
```

Comment:
Remote these as well, they are helpful guides for plugin development, but not relevant to this particular plugin.